hazelcast-sql/src/main/java/com/hazelcast/sql/impl/expression/FieldAccessExpression.java

```diff
@@ -76,33 +74,29 @@ public T eval(Row row, ExpressionEvalContext context) {
     @Override
     public T eval(final Row row, final ExpressionEvalContext context, boolean useLazyDeserialization) {
         // Use lazy deserialization for nested queries. Only the last access should be eager.
-        final Object res = ref.eval(row, context, true);
-        if (res == null) {
+        Object result = ref.eval(row, context, true);
```

Comment:
nit: `result` should be final

Reply:
I consider explicitly marking _effectively final_ variables as `final` a matter of taste. I am restoring `final` in this particular case since the method is long enough.

Reply:
these are not only _effectively final_ (which is not that important because compiler can sort this out) but _intentionally final_, and that is much more important for readability and avoiding some mistakes